pkg/controllers/networktopologyaware/topoautodiscovery/topo_auto_discovery_controller.go

```diff
@@ -0,0 +1,190 @@
+/*
```

Comment:
IMO, add a hyperNode controller is a better way, and then add an auto conversion manager.